blackjax/vi/fullrank_vi.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2020- The Blackjax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Callable, NamedTuple, Tuple
+
+import math
+import jax
+from jax import lax
+import jax.numpy as jnp
+import jax.scipy as jsp
+from jax.flatten_util import ravel_pytree
+from optax import GradientTransformation, OptState
+
+from blackjax.types import PRNGKey, PyTree, Array
+
+__all__ = ["FullrankVIState", "FullrankVIInfo", "sample", "generate_fullrank_logdensity", "step"]
+
+
+def _real_vector_to_cholesky(X):
```

Comment:
Is this jitable? I think we might need to make `m` and `n` as kwarg, and create a closure below when we are setting up the parameters.

Reply:
I am not sure if this is jitable but jax seems to be happy with it. I didn't see any warning.